### PR TITLE
heatseeker: 1.5.1 -> 1.7.1

### LINKED ISSUE #42: Use `/usr/bin/env` to find `stty`
Some distros don't necessarily have `stty` at `/bin/stty`, but `/usr/bin/env` is slightly more standardized.

(I'd offer a patch but I'm currently a total Rust illiterate .)

### PR DESCRIPTION
**Pull request by michaelpj**
The old version is broken on our version of Rust.

Needed patching to fix https://github.com/rschmitt/heatseeker/issues/42.

Also get rid of `legacyCargoFetcher` as desired by
https://github.com/NixOS/nixpkgs/issues/79975.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)https://github.com/NixOS/nixpkgs/issues/79975
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/heatseeker/default.nix [hunk 1]
```diff
@@ -1,28 +1,34 @@
...
 
-  # Delete this on next update; see #79975 for details
-  legacyCargoFetcher = true;
+  buildDepends = [ coreutils ];
 
-  cargoSha256 = "0m3sxbz1iii31s30cnv1970i1mwfhl6gm19k8wv0n7zji30ayx07";
+  patchPhase = ''
+    # https://github.com/rschmitt/heatseeker/issues/42
+    # I've suggested using `/usr/bin/env stty`, but doing that isn't quite as simple
+    # as a substitution, and this works since we have the path to coreutils stty.
```
Comment (bhipple) [inline pkgs/tools/misc/heatseeker/default.nix]: I'd move the comment out of the patchPhase so it isn't exposed to the builder and part of the hash (e.g., put the comment on line 17).  As it stands, a change to the comment will cause a hash change and rebuild of the package.

File: pkgs/tools/misc/heatseeker/default.nix [hunk 2]
```diff
@@ -1,28 +1,34 @@
...
+  cargoSha256 = "0jnlcm7v29m4nc318qgf7r7jvs80s7n04fw83imm506vwr9rxbx9";
 
   # some tests require a tty, this variable turns them off for Travis CI,
   # which we can also make use of
   TRAVIS = "true";
 
-  meta = with stdenv.lib; {
+  meta = with lib; {
     description = "A general-purpose fuzzy selector";
     homepage = https://github.com/rschmitt/heatseeker;
```
Comment (bhipple) [inline pkgs/tools/misc/heatseeker/default.nix]: Might as well also quote this while you're here, for https://github.com/NixOS/rfcs/pull/45, though the nixpkgs-update bot is soon to start doing this automatically if a human doesn't (https://github.com/ryantm/nixpkgs-update/pull/146).

**Review by bhipple — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/heatseeker/default.nix (hunk 1 above)
Comment (michaelpj) [reply pkgs/tools/misc/heatseeker/default.nix]: I don't expect that to happen much, but thanks for the tip!

**Comment by michaelpj** [comment]
Comments addressed, thanks.

**Pull request closed without merging**